EZYVideoPlayer/Classes/EZYVideoPlayer.swift

```diff
@@ -0,0 +1,109 @@
+//
+//  EZYVideoPlayer.swift
+//  EZYVideoPalyer
+//
+//  Created by Shashank Pali on 16/11/22.
+//
+
+import UIKit
+import AVFoundation
+import AVKit
+
+@IBDesignable public class EZYVideoPlayer: UIView, EZYVideoPlayerProtocol {
+    
+    @IBInspectable var title: String = "Video title will be showen here"
+    @IBInspectable var videoURL: String = "http://qthttp.apple.com.edgesuite.net/1010qwoeiuryfg/sl.m3u8"
+    
+    public weak var delegate: EZYVideoPlayerDelegate?
+    private weak var avPlayerLayer: AVPlayerLayer?
+    private var model: EZYVideoPlayerModelProtocol?
+    
+    private weak var overlayView : EZYOverlayProtocol?
+    
+    public override func awakeFromNib() {
+        super.awakeFromNib()
+    }
+    
+    public func startWith(trailerURL: String, thumbnail: UIImage, mute: Bool) {
+        setupPlayer(url: trailerURL)
+        self.layoutSubviews()
+    }
+    
+    public func startWith(mainURL: String, title: String) {
+        setupPlayer(url: mainURL)
+        self.title = title
+        setupComponents()
+    }
+    
+    public override func layoutSubviews() {
+        super.layoutSubviews()
+        avPlayerLayer?.frame = self.bounds
+    }
+    
+    private func setupPlayer(url: String) {
+        overlayView?.removeInstance()
+        avPlayerLayer?.removeFromSuperlayer()
+        model = nil
+        
+        model = EZYVideoPlayerModel(urlString: url)
+        model?.delegate = self
+        guard let player = model?.player else {return}
+        
+        let pl = AVPlayerLayer(player: player)
+        avPlayerLayer = pl
+        avPlayerLayer?.videoGravity = .resizeAspect
+        avPlayerLayer?.backgroundColor = UIColor(white: 0, alpha: 1).cgColor
+        
+        guard (avPlayerLayer != nil) else { return }
+        layer.addSublayer(avPlayerLayer!)
+    }
+    
+    private func setupComponents() {
+        let o = EZYOverlayView()
+        overlayView = o
```

Comment:
If we do so the xcode will show below warnings
- Instance will be immediately deallocated because property 'overlayView' is 'weak'
- Weak reference will always be nil because the referenced object is deallocated here

Which will not be the case I believe it's a bug from xcode. Hence to avoid the warnings I did it like that.